### PR TITLE
Add a parser to parse IEV Excel file

### PR DESCRIPTION
**Pull request by abunashir**
This commit adds a simplified IEV Excel parser, and it also adds a CLI interface that will take IEV Excel file and then it will parse and write those terms in the user provided directory. For now it only handles the basic attributes, but we will add more attributes as we move further. Usages:

```sh
./exe/iev-termbase xlsx2yaml FILE.xlsx --output ./tmp
```

### CODE REVIEW

File: lib/iev/termbase/term_builder.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          #
+          abbrev: nil,
+          alt: nil,
+          example: nil,
+          country_code: nil,
+          review_indicator: nil,
+          authoritative_source_similarity: nil,
+          lineage_source: nil,
+          date_amended: nil,
+          review_date: nil,
```
Comment (ronaldtse) [inline lib/iev/termbase/term_builder.rb:59]: Use PUBLICATIONDATE.

File: lib/iev/termbase/term_builder.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          review_status: nil,
+          review_type: nil,
+          review_decision_date: nil,
+          review_decision_event: nil,
```
Comment (ronaldtse) [inline lib/iev/termbase/term_builder.rb:63]: Since these are all published, write `published`.

File: lib/iev/termbase/term_builder.rb [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          example: nil,
+          country_code: nil,
+          review_indicator: nil,
+          authoritative_source_similarity: nil,
+          lineage_source: nil,
+          date_amended: nil,
+          review_date: nil,
+          review_status: nil,
+          review_type: nil,
+          review_decision_date: nil,
```
Comment (ronaldtse) [inline lib/iev/termbase/term_builder.rb:62]: Use PUBLICATIONDATE.

File: lib/iev/termbase/terminology_sheet.rb [hunk 4]
```diff
@@ -1,108 +1,105 @@
 require_relative "metadata_section"
 require_relative "terms_section"
 
-module Tc211::Termbase
+module Iev::Termbase
```
Comment (ronaldtse) [inline lib/iev/termbase/terminology_sheet.rb:4]: `TerminologySheet` was used to parse ISO TC 211 excel and is not relevant here. In ISO TC 211 excel, every language is on a separate sheet, and each sheet contained a separate metadata and terms section.

All language terms on IEV excel are on the same sheet, and there is no metadata section to parse.

**Comment by ronaldtse** [comment]
From https://gist.github.com/strogonoff/01a6228537a5d5a4762e4d15ce7edaae#file-termtype-ts-L97 , here's a sample:

```yaml
term: abstract test module
termid: 5
eng:
  id: 5
  definition: set of related abstract test cases
  language_code: eng
  notes:
    - Abstract test modules may be nested in a hierarchical way.
  examples: []
  terms:
    # Primary designation
    - designation: abstract test module
      type: expression
      normativeStatus: preferred
      partOfSpeech: noun
      gender: masculine
      grammaticalNumber: mass
    # Synonym 1
    - type: expression
      designation: ATM
      partOfSpeech: noun
      isAbbreviation: true
    # Synonym 2
    - type: symbol
      designation: Δ
  usageInfo: in requirements
  domain: UML
```

We need to do:

* Instead of a `Term` object, let's rename it to `Designation`.

* `normativeStatus`: `preferred` or `deprecated`
* `usageInfo`: all the text between `&lt;...&gt;` in `*ATTRIBUTE` columns (e.g. `&lt;of an item&gt;`
* `geographicalArea`
* `Prefix`: These all mean `prefix`: `Präfix`, `prefix`, `préfixe`, `接尾語`, `접두사`, `przedrostek`, `prefixo`, `词头` (Note that for `pl` and `pt` it is extracted from the definition)
Screen Shot 2020-04-06 at 2 11 27 AM

* Grammar parsing (`partOfSpeech`: `noun` ..., `gender`:... see Gist)

** `adjective` can be: `Adjektiv`, `adj`, `adj`, `形容詞`, `형용사`, `adj`, `形容词`, `(صفة)` (Note that for `ar`, `pt` it is extracted from definition)
Screen Shot 2020-04-06 at 2 14 49 AM

** `noun`: `名詞` (somehow only `jp` has this)
Screen Shot 2020-04-06 at 2 19 12 AM

** `verb`: `動詞` (somehow only `jp` has this)
Screen Shot 2020-04-06 at 2 19 59 AM

* Detect abbreviations: `*ATTRIBUTE` column: `Abkürzung`, `(akronim)`, `(abreviatura inglesa)` (Notice in `pl` `(akronim)` and `pt` `(abreviatura inglesa)` they are embedded in the definition (which we have to extract).
Screen Shot 2020-04-06 at 2 23 11 AM

* Support `SYMBOL` as a separate `Designation` (see YAML sample)

File: lib/iev/termbase/term_builder.rb [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      def find_value_for(key)
+        data.fetch(indices[key], nil)
+      end
+
+      def build_term_object
+        Iev::Termbase::Term.new(
+          id: find_value_for("IEVREF"),
+          entry_status: find_value_for("STATUS"),
+          classification: find_value_for("SYNONYM1STATUS"),
+          authoritative_source: find_value_for("SOURCE"),
```
[the remark this answers is not in the document]
Comment (abunashir) [reply lib/iev/termbase/term_builder.rb]: I've added the source parsing through relaton.

**Comment by abunashir** [comment]
Screen Shot 2020-04-06 at 5 34 58 PM

@ronaldtse, @strogonoff :  This how it looks so far, please let me know if you guys can use it already, and I will keep refining the remaining attributes.

**Comment by ronaldtse** [comment]
Latest screenshot from @abunashir 
Screen Shot 2020-04-07 at 1 02 36 PM

**Comment by abunashir** [comment]
Screen Shot 2020-04-07 at 5 41 10 PM

@ronaldtse: This is the latest update based on your feedback, let me know if we should add anything else there.

**Comment by abunashir** [comment]
Please feel free to merge it unless there are something you would like me to change.

File: lib/iev/termbase/term_builder.rb [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          notes: extract_node_value,
+          definition: extract_definition_value,
+          language_code: three_char_code(find_value_for("LANGUAGE")),
+
+          # @todo: Unsorted Attributes
+          #
+          # We need to revisit this attributes and also update
+          # the correct mapping with the existing attributes.
+          #
+          abbrev: nil,
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/iev/termbase/term_builder.rb:57]: Will be doing in #8.

**Pull request merged**